lab_3_generate_by_ngrams/main.py

```diff
@@ -93,6 +119,22 @@ def encode(self, text: str) -> Optional[tuple[int, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str):
+            return None
+        tokenized_text = self._tokenize(text)
+        if not tokenized_text:
+            return None
+        for symbol in tokenized_text:
+            self._put(symbol)
+        encoded_corpus = []
+        for symbol in tokenized_text:
```

Comment:
Why do you do this in the second `for`? Let's try to opmtimize and do it one loop

Reply:
merged